src/commands/path/path.generate.ts

```diff
@@ -0,0 +1,214 @@
+  } else if (category === dataCategories.SATELLITE_IMAGERY) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}_${gsd}/rgb/${epsg}/`;
+  } else if ([dataCategories.DEM, dataCategories.DSM].includes(category)) {
+    return `s3://${targetBucketName}/${region}/${name}_${date}/${category}_${gsd}/${epsg}/`;
+  } else {
+    throw new Error(`Path Can't be generated from collection as no matching category: ${category}.`);
+  }
+}
+
+/**
+ * Generates specific dataset name based on metadata inputs
+ *
+ * @param {string} region
+ * @param {string} geospatial_description
+ * @param {string} event
+ * @returns {string}
+ */
+export function generateName(region: string, geospatial_description: string, event: string): string {
+  let name = region;
+  if (geospatial_description) {
+    name = geospatial_description.toLowerCase().replace(/\s+/g, '-');
+  }
+  if (event) {
+    name = `${name}-${event.toLowerCase().replace(/\s+/g, '-')}`;
+  }
+  return name;
+}
+
+export function getCategory(collection: StacCollection): string {
+  const category = collection['linz:geospatial_category'] as string;
+  if (!category) {
+    throw new Error('No category in collection');
+  }
+  return category;
+}
+
+export function getGeographicDescription(collection: StacCollection): string {
+  // This is optional metadata, therefore returning nothing is ok.
+  return collection['linz:geographic_description'] as string;
+}
+
+export function getEvent(collection: StacCollection): string {
+  // This is optional metadata, therefore returning nothing is ok.
+  return collection['linz:event_name'] as string;
+}
+
+export function getRegion(collection: StacCollection): string {
+  const region = collection['linz:region'] as string;
+  if (!region) {
+    throw new Error('No region in collection');
+  } else if (!regions.includes(region)) {
+    throw new Error(`Invalid region: ${region}`);
+  }
+  return region;
+}
+
+export function getDate(collection: StacCollection): string {
+  const interval = collection.extent.temporal.interval[0];
+  const startYear = interval[0]?.slice(0, 4);
+  const endYear = interval[1]?.slice(0, 4);
+
+  if (!startYear || !endYear) {
+    throw new Error(`Missing datetime in interval: ${interval}`);
+  }
+  if (startYear === endYear) {
+    return startYear;
+  }
+  return `${startYear}-${endYear}`;
+}
+
+/*
+ *  nb: The following functions: 'loadFirstTiff', 'extractGsd', and 'extractEpsg' are workarounds
+ *  to manage the block information from being added to the collection (eo stac extension).
+ *  Once this is fixed the following functions can be replaced.
+ */
+
+/**
+ * Gets first item & tiff listed in collection
+ *
+ * @async
+ * @param {string} source
+ * @param {StacCollection} collection
+ * @returns {Promise<CogTiff>}
+ */
+export async function loadFirstTiff(source: string, collection: StacCollection): Promise<CogTiff> {
+  const itemLink = collection.links[2]?.href.replace('./', ''); // [2] to skip root & self links
```

Comment:
safer to find the first item `collection.links.find(f => f.rel === 'item')`

it is much safer to use URLS, especially if they have things like `../foo.tiff` or even just missing `"foo.tiff"`

`new URL(link, source).href`